code/modules/reagents/chemistry/machinery/reagentgrinder.dm

```diff
@@ -433,7 +435,7 @@
 			if (O.Uses > 0)
 				beaker.reagents.add_reagent("slimejelly",min(20, space))
 			remove_object(O)
-		else if(istype(I, /obj/item/reagent_containers))
+		if(istype(I, /obj/item/reagent_containers))
```

Comment:
this one too

Reply:
No, this one fixes #31697. They actually were all ifs before kevinz's refactor. If a /obj/item/reagent_containers/food/snacks subtype is not handled at #L377 it's supposed to be handled by the generic reagent container code, instead of not being handled at all

Reply:
And yeah, grinders are shitcode